includes/blocks.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "constants.h"
+
+class Block {
```

Comment:
You will have to explain to me the necessity of adding these two new classes sometime so I can better understand the design arch that you're trying to go for.